src/Gui/DockWindowManager.cpp

```diff
@@ -380,6 +380,20 @@ void DockWindowManager::setup(DockWindowItems* items)
         }
     }
 
+    // Don't always tabify after switching the workbench
+    static bool tabify = false;
+    if (!tabify) {
+        // tabify dock widgets for which "tabbed" is true and which have the same position
+        for (int i=0; i<4; i++) {

```

Comment:
Would that make sense to iterate up to `areas->length()` just in case it evolves in the future ?

Reply:
Not possible because areas is a C-array of QList's: `QList<QDockWidget*> areas[4];`
An option is to use a std::array instead.

Reply:
Right. `areas` is a local variable so we shouldn't expect much problems when changing its type. Let you see what you think is best.

Reply:
Other solution : how about using a `constexpr` of `log2(Qt::AllDockWidgetAreas+1)` to define the array size and so the loop range ?

Reply:
Replaced the C array with std::array. Also refactored the function setup() as it was becoming too long.